### ISSUE TITLE
Cannot use 'import.meta' outside a module

### ISSUE THREAD

**Issue by kevinmarrec** [webpro/knip#536]
Hi, i'm trying to use Knip but it seems we can't use `import.meta` with it.
I'm using Vitest and here is the `vitest.config.ts` file i'm using :

```ts
import path from 'node:path'

import { defineConfig } from 'vitest/config'

export default defineConfig({
  resolve: {
    alias: {
      '~': path.resolve(import.meta.dirname, 'src'),
    },
  },
  test: {
    coverage: {
      enabled: true,
      include: ['src'],
      thresholds: {
        statements: 100,
      },
    },
  },
})
```

And the error I get when running Knip :
![Screenshot 2024-03-09 114524](https://github.com/webpro/knip/assets/25272043/ebfcb7bd-0358-4eeb-8c20-435d338ae4aa)

All my `package.json` files are using `"type": "module"`, and I don't want to be forced to use `__dirname` (CommonJS) instead of `import.meta.dirname` (ESM with Node.js 20+)

**Comment by webpro** [comment]
Looks like this known issue: https://knip.dev/reference/known-issues#ts-config-files-using-esm-features

So a rename to `.js` could be a work-around.

**Comment by kevinmarrec** [comment]
Thanks, sorry it seems I didn't search enough in the documentation.

I tried the `ignore` solution before seeing this, but without any luck because I think it still load it then ignore it in reports. I'll try with negated `project` patterns.

If issue is known and documented, maybe it's relevant to close here.

**Comment by webpro** [comment]
Good to know the `ignore` pattern doesn't work, I'll remove it (didn't like the option anyway).  Even if it gets you through, you'd miss the plugin functionality and end up with false positives. Will keep this ticket open until I've updated docs.

The only actual solution for now is to rename, not such a huge deal I guess.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.1.0. See [Release 5.1.0](https://github.com/webpro/knip/releases/tag/5.1.0) for release notes._
